pkg/api/component_report.go

```diff
@@ -1660,21 +1661,73 @@ func (c *componentReportGenerator) assessComponentStatus(sampleTotal, sampleSucc
 	status := apitype.MissingBasis
 	fischerExact := 0.0
 	if baseTotal != 0 {
-		if sampleTotal == 0 {
+		// if the unadjusted sample was 0 then nothing to do
+		if initialSampleTotal == 0 {
 			if c.IgnoreMissing {
 				status = apitype.NotSignificant
 
 			} else {
 				status = apitype.MissingSample
 			}
 		} else {
-			if c.MinimumFailure != 0 && (sampleTotal-sampleSuccess-sampleFlake) < c.MinimumFailure {
-				return apitype.NotSignificant, fischerExact
+			// see if we had a significant regression prior to adjusting
```

Comment:
I wonder if it is more readable if we create a common function to do the calculation. The function will be called twice, one for preAdjustment and one for postAdjustment, and then we can compare the result to decide on the report value.

Reply:
I started out thinking I would.  But then each case seemed to have it's own prechecks and the common logic seemed to come down to

```
fischer.FisherExactTest(baseTotal-baseSuccess-baseFlake,
					baseSuccess+baseFlake,
					sampleTotal-sampleSuccess-sampleFlake,
					sampleSuccess+sampleFlake)
				significant = r < 1-float64(c.Confidence)/100
				fischerExact = r
```
Didn't seem worth the additional refactor but I could if you think I should.  Maybe more would fall out too...